From 102adc148612b50a4e3f12a88871ddc47bfb469e Mon Sep 17 00:00:00 2001
From: David Malcolm <dmalcolm@redhat.com>
Date: Thu, 2 Sep 2021 17:17:50 -0400
Subject: [PATCH 17/17] Conditionalize test for PR libstdc++/71181 on
 __LIBSTDCXX_SO_VERSION >= 7

This test was added upstream 2016-06-20 as part of:
  re PR libstdc++/71181 (Reserving in unordered_map doesn't reserve enough)
    https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=29dbb034cb3199167a9d0aaed040733c72326eed

but fails when run in DTS against a system libstdc++.so from an older GCC.
---
 .../testsuite/23_containers/unordered_set/hash_policy/71181.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libstdc++-v3/testsuite/23_containers/unordered_set/hash_policy/71181.cc b/libstdc++-v3/testsuite/23_containers/unordered_set/hash_policy/71181.cc
index 324d735b8..b2dcd0137 100644
--- a/libstdc++-v3/testsuite/23_containers/unordered_set/hash_policy/71181.cc
+++ b/libstdc++-v3/testsuite/23_containers/unordered_set/hash_policy/71181.cc
@@ -39,7 +39,10 @@ template<typename _USet>
 
 	us.insert(nb_insert);
 
+	// Fixed upstream in GCC 7
+#if __LIBSTDCXX_SO_VERSION >= 7
 	VERIFY( us.bucket_count() == bkts );
+#endif
       }
   }
 
-- 
2.31.1

